[p0] CI broken on main: AppDelegateTests.swift fails to compile (actor isolation)

CI on `main` (HEAD `9bf1a3b`) fails on the `Build & Test` job. The latest 3 main runs all failed/cancelled.

## Error
```
Tests/EyePostureReminderTests/Services/AppDelegateTests.swift:102:38: error:
  call to main actor-isolated instance method 'settings(for:)' in a synchronous nonisolated context
  snapshot: { settings.settings(for: .eyes) },
```

The `SettingsClient.snapshot` closure is `@Sendable () -> ReminderSettings`. It captures a `@MainActor SettingsStore` and calls a `@MainActor` instance method synchronously, which the CI Swift compiler rejects under strict concurrency. Local Xcode 26.4 is more permissive and lets it through, masking the regression introduced by PR #699 and friends.

## Blast radius
- Every PR rebased on main fails `Build & Test` on this exact compile error (e.g. PR #703, PR #704).
- `./scripts/build.sh all` is wedged in CI.

## Fix
In `AppDelegateTests.makeAppFeatureStore`, capture the snapshot value once on the main actor (the test class is already `@MainActor`) and return that captured value from the closure.

## Evidence
- Failing job: https://github.com/yashasg/fantastic-octo-fortnight/actions/runs/25882666516
- Workflow run on main #700: failure
- Workflow run on main #699: cancelled
- Workflow run on main #696: cancelled

--- a/EyePostureReminder/App/AppDelegate.swift
+++ b/EyePostureReminder/App/AppDelegate.swift
@@ -98,13 +98,11 @@ final class AppDelegate: NSObject, UIApplicationDelegate, UNUserNotificationCent
     /// prevents `TrueInterruptSkippedBanner` from ever rendering on the first
     /// cold launch (#457).
     ///
-    /// Also pre-seeds `hasSeenOnboarding` so the TCA root state seeded in
-    /// `EyePostureReminderApp.init()` (`p0-tca-11` / #674) sees the correct
-    /// value. Without this, `--reset-onboarding` and `--skip-onboarding` are
-    /// applied too late (in `didFinishLaunching`) and the store keeps the
-    /// stale value left over from the previous test launch (#707).
+    /// `hasSeenOnboarding` is pre-seeded earlier — directly in
+    /// `EyePostureReminderApp.init()` — because `@UIApplicationDelegateAdaptor`
+    /// only instantiates this delegate as part of `UIApplicationMain`, which
+    /// runs *after* the App struct's `init()` body returns. See #707.
     private func preSeedUITestDefaults() {
-        preSeedHasSeenOnboardingIfNeeded()
         if launchArguments.contains("--simulate-screen-time-not-determined") {
             uiTestDefaults.set(
                 ScreenTimeAuthorizationStatus.notDetermined.rawValue,
@@ -123,31 +121,6 @@ final class AppDelegate: NSObject, UIApplicationDelegate, UNUserNotificationCent
                !launchArguments.contains("--dismiss-true-interrupt-banner") {
                 uiTestDefaults.set(false, forKey: AppStorageKey.trueInterruptSkippedBannerDismissed)
             }
-        }
-    }
-
-    /// Synchronizes the `hasSeenOnboarding` UserDefaults key with the
-    /// onboarding-affecting launch args before `EyePostureReminderApp.init()`
-    /// reads it to seed `AppFeature.State.hasSeenOnboarding` (#707).
-    ///
-    /// `--reset-onboarding` removes the key; the four "skip past onboarding"
-    /// launch args (used by tests that need to start on Home/overlay screens)
-    /// set it to `true`. The full `applyUITestLaunchArguments()` pass still
-    /// runs from `didFinishLaunching` and handles `SettingsStore` resets and
-    /// overlay-type seeding — this hook only mirrors the onboarding gate so
-    /// the TCA store starts on the correct branch.
-    private func preSeedHasSeenOnboardingIfNeeded() {
-        if launchArguments.contains("--reset-onboarding") {
-            uiTestDefaults.removeObject(forKey: AppStorageKey.hasSeenOnboarding)
-            return
-        }
-        let skipsOnboarding =
-            launchArguments.contains("--skip-onboarding") ||
-            launchArguments.contains("--show-overlay-eyes") ||
-            launchArguments.contains("--show-overlay-posture") ||
-            launchArguments.contains("--simulate-screen-time-not-determined")
-        if skipsOnboarding {
-            uiTestDefaults.set(true, forKey: AppStorageKey.hasSeenOnboarding)
         }
     }
 #endif

--- a/EyePostureReminder/App/AppDelegate.swift
+++ b/EyePostureReminder/App/AppDelegate.swift
@@ -97,7 +97,14 @@ final class AppDelegate: NSObject, UIApplicationDelegate, UNUserNotificationCent
     /// and falls back to `ScreenTimeAuthorizationNoop(.unavailable)`, which
     /// prevents `TrueInterruptSkippedBanner` from ever rendering on the first
     /// cold launch (#457).
+    ///
+    /// Also pre-seeds `hasSeenOnboarding` so the TCA root state seeded in
+    /// `EyePostureReminderApp.init()` (`p0-tca-11` / #674) sees the correct
+    /// value. Without this, `--reset-onboarding` and `--skip-onboarding` are
+    /// applied too late (in `didFinishLaunching`) and the store keeps the
+    /// stale value left over from the previous test launch (#707).
     private func preSeedUITestDefaults() {
+        preSeedHasSeenOnboardingIfNeeded()
         if launchArguments.contains("--simulate-screen-time-not-determined") {
             uiTestDefaults.set(
                 ScreenTimeAuthorizationStatus.notDetermined.rawValue,
@@ -116,6 +123,31 @@ final class AppDelegate: NSObject, UIApplicationDelegate, UNUserNotificationCent
                !launchArguments.contains("--dismiss-true-interrupt-banner") {
                 uiTestDefaults.set(false, forKey: AppStorageKey.trueInterruptSkippedBannerDismissed)
             }
+        }
+    }
+
+    /// Synchronizes the `hasSeenOnboarding` UserDefaults key with the
+    /// onboarding-affecting launch args before `EyePostureReminderApp.init()`
+    /// reads it to seed `AppFeature.State.hasSeenOnboarding` (#707).
+    ///
+    /// `--reset-onboarding` removes the key; the four "skip past onboarding"
+    /// launch args (used by tests that need to start on Home/overlay screens)
+    /// set it to `true`. The full `applyUITestLaunchArguments()` pass still
+    /// runs from `didFinishLaunching` and handles `SettingsStore` resets and
+    /// overlay-type seeding — this hook only mirrors the onboarding gate so
+    /// the TCA store starts on the correct branch.
+    private func preSeedHasSeenOnboardingIfNeeded() {
+        if launchArguments.contains("--reset-onboarding") {
+            uiTestDefaults.removeObject(forKey: AppStorageKey.hasSeenOnboarding)
+            return
+        }
+        let skipsOnboarding =
+            launchArguments.contains("--skip-onboarding") ||
+            launchArguments.contains("--show-overlay-eyes") ||
+            launchArguments.contains("--show-overlay-posture") ||
+            launchArguments.contains("--simulate-screen-time-not-determined")
+        if skipsOnboarding {
+            uiTestDefaults.set(true, forKey: AppStorageKey.hasSeenOnboarding)
         }
     }
 #endif

--- a/EyePostureReminder/App/EyePostureReminderApp.swift
+++ b/EyePostureReminder/App/EyePostureReminderApp.swift
@@ -15,12 +15,52 @@ struct EyePostureReminderApp: App {
 
     init() {
         AppTypography.registerFonts()
+#if DEBUG
+        // Mirror onboarding-affecting launch args into UserDefaults BEFORE the
+        // initial state seed below reads them. `AppDelegate.init()` runs too
+        // late under `@UIApplicationDelegateAdaptor` (UIKit only instantiates
+        // the delegate as part of `UIApplicationMain`, after this `App.init`
+        // returns), so deferring the sync to `AppDelegate` left the store on
+        // the stale `true` value left in UserDefaults from the previous test
+        // launch and onboarding tests landed on Home (#707).
+        Self.preSeedHasSeenOnboardingFromLaunchArgsIfNeeded()
+#endif
         var initialState = AppFeature.State()
         initialState.hasSeenOnboarding = UserDefaults.standard.bool(
             forKey: AppStorageKey.hasSeenOnboarding
         )
         self.store = Store(initialState: initialState) { AppFeature() }
     }
+
+#if DEBUG
+    /// Synchronizes the `hasSeenOnboarding` UserDefaults key with the
+    /// onboarding-affecting launch arguments before `init()` reads it to seed
+    /// `AppFeature.State` (#707).
+    ///
+    /// `--reset-onboarding` clears the key; the four "skip past onboarding"
+    /// launch arguments (used by tests that need to start on Home/overlay
+    /// screens) set it to `true`. The full `AppDelegate.applyUITestLaunch
+    /// Arguments()` pass still runs from `didFinishLaunchingWithOptions` and
+    /// handles `SettingsStore` resets and overlay-type seeding — this hook
+    /// only mirrors the onboarding gate so the TCA root state starts on the
+    /// correct branch. `#if DEBUG` keeps the launch-arg backdoor out of
+    /// Release/TestFlight builds (re: #350/#405).
+    private static func preSeedHasSeenOnboardingFromLaunchArgsIfNeeded() {
+        let args = CommandLine.arguments
+        if args.contains("--reset-onboarding") {
+            UserDefaults.standard.removeObject(forKey: AppStorageKey.hasSeenOnboarding)
+            return
+        }
+        let skipsOnboarding =
+            args.contains("--skip-onboarding") ||
+            args.contains("--show-overlay-eyes") ||
+            args.contains("--show-overlay-posture") ||
+            args.contains("--simulate-screen-time-not-determined")
+        if skipsOnboarding {
+            UserDefaults.standard.set(true, forKey: AppStorageKey.hasSeenOnboarding)
+        }
+    }
+#endif
 
     var body: some Scene {
         WindowGroup {

--- a/Tests/EyePostureReminderTests/Services/AppDelegateTests.swift
+++ b/Tests/EyePostureReminderTests/Services/AppDelegateTests.swift
@@ -95,11 +95,20 @@ final class AppDelegateTests: XCTestCase {
         settings: SettingsStore,
         snoozeCountWrites: LockIsolated<[Int]>
     ) -> StoreOf<AppFeature> {
-        Store(initialState: AppFeature.State()) {
+        // The `snapshot` closure on `SettingsClient` is `@Sendable` and therefore
+        // cannot synchronously call `@MainActor`-isolated methods on
+        // `SettingsStore`. Capture the eyes snapshot once on the main actor (this
+        // function is invoked from a `@MainActor` test setUp) and return the
+        // captured value. The `SchedulingFeature` paths under test never re-read
+        // `snapshot`, so a stable initial value is sufficient.
+        let initialEyesSnapshot = MainActor.assumeIsolated {
+            settings.settings(for: .eyes)
+        }
+        return Store(initialState: AppFeature.State()) {
             AppFeature()
         } withDependencies: {
             $0.settingsClient = SettingsClient(
-                snapshot: { settings.settings(for: .eyes) },
+                snapshot: { initialEyesSnapshot },
                 stream: { .finished },
                 updateGlobalEnabled: { _ in },
                 updateEyesEnabled: { _ in },

